BUILD.gn

```diff
@@ -231,6 +231,14 @@ if (current_toolchain != "${dir_pw_toolchain}/default:default") {
         ]
       }
     }
+
+    if (getenv("JAVA_PATH") != "") {
+      deps += [
+        "${chip_root}/examples/java-matter-controller:java-matter-controller",
+      ]
+      chip_build_tools = false
+      chip_enable_python_modules = false
```

Comment:
This cause the following error during build

```
ERROR at //BUILD.gn:240:36: Assignment had no effect.
      chip_enable_python_modules = false
                                   ^----
You set the variable "chip_enable_python_modules" here and it was unused before it went
out of scope.
['gn', 'gen', '/usr/local/google/home/yufengwang/Workspace/yunhanw/connectedhomeip/.environment/gn_out', '--args=dir_pigweed="/usr/local/google/home/yufengwang/Workspace/yunhanw/connectedhomeip/third_party/pigweed/repo"']
```

Reply:
thanks for pointing out this issue, actually we don't need set this variable since we have set chip_build_tool as false, then the python would not be built as well. just remove this line.

Reply:
I think we also don't need to set 'chip_build_tools' ? There is another error

```
ERROR at //BUILD.gn:239:26: Assignment had no effect.
      chip_build_tools = false
                         ^----
You set the variable "chip_build_tools" here and it was unused before it went
out of scope.

```

Reply:
yes, just removed.